File: src/main/java/com/ciazhar/authserver/model/jpa/Role.java
package com.ciazhar.authserver.model.jpa;

import org.hibernate.annotations.GenericGenerator;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;

/**
 * Created by ciazhar on 5/27/17.
 */

/**
 * Model untuk role
 */
@Entity
public class Role {

    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid",strategy = "uuid2")
    @Column(name = "id_role")
    private String id;

    @Column(name = "nama_role", unique = true)
    @NotNull
    private String nama;

    @Column(name = "label_role")
    @NotNull
    private String label;


    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNama() {
        return nama;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

}

File: src/main/kotlin/com/ciazhar/authserver/model/jpa/Role.kt
package com.ciazhar.authserver.model.jpa

import org.hibernate.annotations.GenericGenerator

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.validation.constraints.NotNull

/**
 * Created by ciazhar on 5/27/17.
 */

/**
 * Model untuk role
 */
@Entity
class Role {

    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(name = "id_role")
    var id: String? = null

    @Column(name = "nama_role", unique = true)
    @NotNull
    var nama: String? = null

    @Column(name = "label_role")
    @NotNull
    var label: String? = null

}
